lib/sprockets/sass/importer.rb

```diff
@@ -30,65 +35,113 @@ def mtime(path, options)
       end
 
       # @see Sass::Importers::Base#key
-      def key(path, options)
+      def key(path, _options)
         path = Pathname.new(path)
         ["#{self.class.name}:#{path.dirname.expand_path}", path.basename]
       end
 
       # @see Sass::Importers::Base#to_s
       def to_s
-        "#{self.class.name}:#{context.pathname}"
+        inspect
       end
 
-      protected
+    protected
 
       # Create a Sass::Engine from the given path.
       def engine_from_path(path, base_path, options)
         context = options[:custom][:sprockets_context]
-        pathname = resolve(context, path, base_path) or return nil
+        (pathname = resolve(context, path, base_path)) || (return nil)
         context.depend_on pathname
         ::Sass::Engine.new evaluate(context, pathname), options.merge(
-          :filename => pathname.to_s,
-          :syntax   => syntax(pathname),
-          :importer => self
+          filename: pathname.to_s,
+          syntax: syntax(pathname),
+          importer: self,
+          custom: { sprockets_context: context }
         )
       end
 
       # Create a Sass::Engine that will handle importing
       # a glob of files.
       def engine_from_glob(glob, base_path, options)
         context = options[:custom][:sprockets_context]
-        imports = resolve_glob(context, glob, base_path).inject('') do |imports, path|
+        engine_imports = resolve_glob(context, glob, base_path).reduce('') do |imports, path|
           context.depend_on path
           relative_path = path.relative_path_from Pathname.new(base_path).dirname
           imports << %(@import "#{relative_path}";\n)
         end
-        return nil if imports.empty?
-        ::Sass::Engine.new imports, options.merge(
-          :filename => base_path.to_s,
-          :syntax   => :scss,
-          :importer => self
+        return nil if engine_imports.empty?
+        ::Sass::Engine.new engine_imports, options.merge(
+          filename: base_path.to_s,
+          syntax: syntax(base_path.to_s),
+          importer: self,
+          custom: { sprockets_context: context }
         )
       end
 
+      def resolve_path_with_load_paths(context, path, root_path, file)
+        context.resolve(file.to_s, load_paths: context.environment.paths, base_path: root_path, accept: syntax_mime_type(path))
+      rescue
+        nil
+      end
+
       # Finds an asset from the given path. This is where
       # we make Sprockets behave like Sass, and import partial
       # style paths.
       def resolve(context, path, base_path)
-        possible_files(context, path, base_path).each do |file|
-          context.resolve(file) { |found| return found if context.asset_requirable?(found) }
+        paths, root_path = possible_files(context, path, base_path)
+        if Sprockets::Sass.version_of_sprockets >= 3
+          paths.each do |file|
```

Comment:
The resolve method from sprockets was changed and now requires additional parameters ,  as you can see in method ```resolve_path_with_load_paths```  from the Importer.